pkg/component/kubernetes/apiserverexposure/service.go

```diff
@@ -126,6 +126,10 @@ func (s *service) Deploy(ctx context.Context) error {
 	obj := s.emptyService()
 
 	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, s.client, obj, func() error {
+		// TODO(scheererj): Remove this after release of v1.125
+		// Clean up old annotations as unnecessary load balancer annotations may reside on a ClusterIP service.
+		obj.Annotations = map[string]string{}
```

Comment:
Can this have the side effect of reverting back annotation which is set by external component/extension? 
The idea of not overwriting annotations/labels was to preserve an annotation/label which could be contributed by external component.

Reply:
Yes, this has the side effect that all annotations are removed and replaced by the small set of network policy annotations maintained by `gardenlet`.
Are you aware of any component setting annotations on this service?